### PR TITLE
Adding upheno_custom_mapping preview

### PR DESCRIPTION
**Pull request by souzadevinicius**

### CODE REVIEW

File: mappings/upheno_custom_mapping_head_100.sssom.tsv [hunk 1]
```diff
@@ -0,0 +1,101 @@
+subject_id	subject_label	subject_source	predicate_id	object_id	object_label	object_source	mapping_justification
+HP:0010951	Abnormal third ventricle morphology (HPO)	obo:0010951	semapv:crossSpeciesExactMatch	XPO:0102755	abnormal third ventricle morphology (XPO)	obo:0102755	semapv:LexicalMatching
```
Comment (matentzn) [inline mappings/upheno_custom_mapping_head_100.sssom.tsv:2]: - [ ] object_id should always be the ontology (`obo:hp` or `obo:mp`) of the `object_id`
- [ ] Same for subject_id
- [ ] I would like to add `subject_match_field` and `object_match_field` - do you think that would be possible? (https://mapping-commons.github.io/sssom/)

**Comment by matentzn** [comment]
@souzadevinicius please dont forget about this!

File: mappings/upheno_custom_mapping_head_100.sssom.tsv [hunk 2]
```diff
@@ -0,0 +1,101 @@
+subject_id	subject_label	subject_source	predicate_id	object_id	object_label	object_source	mapping_justification
```
Comment (matentzn) [inline mappings/upheno_custom_mapping_head_100.sssom.tsv:1]: 1. use converter.compress_or_standardize() from the curies package for the subject_id and object_id
2. manually curate a yaml file with relevant metadata in this directory: https://github.com/monarch-initiative/monarch-mapping-commons/tree/main/metadata
3. Add the instructions on how to generate this file in https://github.com/monarch-initiative/monarch-mapping-commons/blob/main/monarch_mapping_commons.Makefile

File: mappings/upheno_custom_mapping_head_100.sssom.tsv [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+obo:0000826	abnormal third ventricle morphology (MPO)	obo:mp	semapv:crossSpeciesExactMatch	obo:0010951	Abnormal third ventricle morphology (HPO)	obo:hp	semapv:LexicalMatching
```
Comment (matentzn) [inline mappings/upheno_custom_mapping_head_100.sssom.tsv:2]: subject_id should be HP:0000826

**Review by matentzn — APPROVED** [review]

File: monarch_mapping_commons.Makefile [hunk 4]
```diff
@@ -40,3 +40,13 @@ $(MAPPING_DIR)/umls_hp.sssom.tsv:
 benchmark:
 	pip install py-spy
 	sudo py-spy record -o flamegraph.svg -- $(SSSOM_TOOLKIT) validate $(MAPPING_DIR)/gene_mappings.sssom.tsv
+
+
+$(TMP_DIR)/upheno/%:
+	mkdir -p $(TMP_DIR)/upheno/
+	wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/$* -O $@
+
+$(MAPPING_DIR)/upheno_custom_mapping.sssom.tsv: $(patsubst %, $(TMP_DIR)/upheno/%, upheno_species_lexical.csv upheno_mapping_logical.csv upheno_all_with_relations.owl)
```
Comment (matentzn) [inline monarch_mapping_commons.Makefile:49]: Nice way to reduce redundancy!

Comment (matentzn) [inline monarch_mapping_commons.Makefile:49]:
```suggestion
$(MAPPING_DIR)/upheno_custom.sssom.tsv: $(patsubst %, $(TMP_DIR)/upheno/%, upheno_species_lexical.csv upheno_mapping_logical.csv upheno_all_with_relations.owl)
```

**Comment by matentzn** [comment]
@souzadevinicius the error log is too big to display here but here is the error log:

```
2023-12-08T09:50:10.9121521Z 2023-12-08 09:49:03 (27.2 MB/s) - ‘tmp/upheno/upheno_all_with_relations.owl’ saved [513061413/513061413]
2023-12-08T09:50:10.9121527Z 
2023-12-08T09:50:10.9121725Z bash: line 1: phenio-toolkit: command not found
2023-12-08T09:50:10.9122120Z make[1]: *** [monarch_mapping_commons.Makefile:51: mappings/upheno_custom.sssom.tsv] Error 127
2023-12-08T09:50:10.9122254Z make: *** [Makefile:37: mappings] Error 2
2023-12-08T09:50:10.9123040Z wget https://raw.githubusercontent.com/monarch-initiative/umls-ingest/main/src/umls_ingest/mappings/hp_mesh.sssom.tsv -O mappings/hp_mesh.sssom.tsv
2023-12-08T09:50:10.9123724Z wget https://raw.githubusercontent.com/monarch-initiative/umls-ingest/main/src/umls_ingest/mappings/umls_hp.sssom.tsv -O mappings/umls_hp.sssom.tsv
2023-12-08T09:50:10.9123853Z mkdir -p tmp/upheno/
2023-12-08T09:50:10.9124554Z wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/upheno_species_lexical.csv -O tmp/upheno/upheno_species_lexical.csv
2023-12-08T09:50:10.9124682Z mkdir -p tmp/upheno/
2023-12-08T09:50:10.9125381Z wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/upheno_mapping_logical.csv -O tmp/upheno/upheno_mapping_logical.csv
2023-12-08T09:50:10.9125498Z mkdir -p tmp/upheno/
2023-12-08T09:50:10.9126222Z wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/upheno_all_with_relations.owl -O tmp/upheno/upheno_all_with_relations.owl
2023-12-08T09:50:10.9126343Z mkdir -p mappings tmp
2023-12-08T09:50:10.9126943Z phenio-toolkit lexical-mapping --species-lexical tmp/upheno/upheno_species_lexical.csv -m tmp/upheno/upheno_mapping_logical.csv -o tmp
2023-12-08T09:50:10.9127294Z make[1]: Leaving directory '/__w/monarch-mapping-commons/monarch-mapping-commons'
2023-12-08T09:50:10.9137125Z ##[error]Process completed with exit code 2.
```

**Comment by souzadevinicius** [comment]
> @souzadevinicius the error log is too big to display here but here is the error log:
> 
> ```
> 2023-12-08T09:50:10.9121521Z 2023-12-08 09:49:03 (27.2 MB/s) - ‘tmp/upheno/upheno_all_with_relations.owl’ saved [513061413/513061413]
> 2023-12-08T09:50:10.9121527Z 
> 2023-12-08T09:50:10.9121725Z bash: line 1: phenio-toolkit: command not found
> 2023-12-08T09:50:10.9122120Z make[1]: *** [monarch_mapping_commons.Makefile:51: mappings/upheno_custom.sssom.tsv] Error 127
> 2023-12-08T09:50:10.9122254Z make: *** [Makefile:37: mappings] Error 2
> 2023-12-08T09:50:10.9123040Z wget https://raw.githubusercontent.com/monarch-initiative/umls-ingest/main/src/umls_ingest/mappings/hp_mesh.sssom.tsv -O mappings/hp_mesh.sssom.tsv
> 2023-12-08T09:50:10.9123724Z wget https://raw.githubusercontent.com/monarch-initiative/umls-ingest/main/src/umls_ingest/mappings/umls_hp.sssom.tsv -O mappings/umls_hp.sssom.tsv
> 2023-12-08T09:50:10.9123853Z mkdir -p tmp/upheno/
> 2023-12-08T09:50:10.9124554Z wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/upheno_species_lexical.csv -O tmp/upheno/upheno_species_lexical.csv
> 2023-12-08T09:50:10.9124682Z mkdir -p tmp/upheno/
> 2023-12-08T09:50:10.9125381Z wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/upheno_mapping_logical.csv -O tmp/upheno/upheno_mapping_logical.csv
> 2023-12-08T09:50:10.9125498Z mkdir -p tmp/upheno/
> 2023-12-08T09:50:10.9126222Z wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/upheno_all_with_relations.owl -O tmp/upheno/upheno_all_with_relations.owl
> 2023-12-08T09:50:10.9126343Z mkdir -p mappings tmp
> 2023-12-08T09:50:10.9126943Z phenio-toolkit lexical-mapping --species-lexical tmp/upheno/upheno_species_lexical.csv -m tmp/upheno/upheno_mapping_logical.csv -o tmp
> 2023-12-08T09:50:10.9127294Z make[1]: Leaving directory '/__w/monarch-mapping-commons/monarch-mapping-commons'
> 2023-12-08T09:50:10.9137125Z ##[error]Process completed with exit code 2.
> ```
Does it come from github workflow? 
During the merge process all workflows have been passed.

**Comment by matentzn** [comment]
Yes, because you had not yet added your mapping to the registry.yml file - I did that, now it is tested.

**Comment by souzadevinicius** [comment]
It's weird. Because the phenio-toolkit is published here:
https://pypi.org/project/phenio-toolkit/

Additionally, it is declared to be installed in toml file:

https://github.com/monarch-initiative/monarch-mapping-commons/blob/21b412983178b95aa6fe6e626b8985ebd519971d/pyproject.toml#L15

Could you try to clone it on your local machine?

**Comment by matentzn** [comment]
Cant now, toooo busy; my guess: the GitHub action running the test is running on ODK and does not install poetry dependencies. Maybe you need to add a pip install step in the action?

**Review by matentzn — APPROVED** [review]

File: monarch_mapping_commons.Makefile [hunk 5]
```diff
@@ -40,3 +40,13 @@ $(MAPPING_DIR)/umls_hp.sssom.tsv:
...
 	sudo py-spy record -o flamegraph.svg -- $(SSSOM_TOOLKIT) validate $(MAPPING_DIR)/gene_mappings.sssom.tsv
+
+
+$(TMP_DIR)/upheno/%:
+	mkdir -p $(TMP_DIR)/upheno/
+	wget https://bbop-ontologies.s3.amazonaws.com/upheno/current/upheno-release/all/$* -O $@
+
+$(MAPPING_DIR)/upheno_custom.sssom.tsv: $(patsubst %, $(TMP_DIR)/upheno/%, upheno_species_lexical.csv upheno_mapping_logical.csv upheno_all_with_relations.owl)
+	mkdir -p $(MAPPING_DIR) $(TMP_DIR)
+	phenio-toolkit lexical-mapping --species-lexical $(TMP_DIR)/upheno/upheno_species_lexical.csv -m $(TMP_DIR)/upheno/upheno_mapping_logical.csv -o $(TMP_DIR)
```
Comment (anitacaron) [inline monarch_mapping_commons.Makefile:51]: It would be good to add the poetry run variable `$(RUN)` here too. I'm adding it to my PR.

**Pull request merged**